x/incentives/types/msgs.go

```diff
@@ -55,8 +56,32 @@ func (m MsgCreateGauge) ValidateBasic() error {
 		return errors.New("distribution period should be 1 epoch for perpetual gauge")
 	}
 
-	if lockuptypes.LockQueryType_name[int32(m.DistributeTo.LockQueryType)] != "ByDuration" {
-		return errors.New("only duration query condition is allowed. Start time distr conditions is an obsolete codepath slated for deletion")
+	if lockType == lockuptypes.ByTime {
+		return errors.New("start time distr conditions is an obsolete codepath slated for deletion")
+	}
+
+	if isNoLockGauge {
+		if m.PoolId == 0 {
+			return errors.New("pool id should be set for no lock distr condition")
+		}
+
+		if m.DistributeTo.Denom != "" {
+			return errors.New(`no lock gauge denom should be unset. It will be automatically set to the NoLockExternalGaugeDenom(<pool id>)
+			 format internally, allowing for querying the gauges by denom with this prefix`)
+		}
+
+		if m.DistributeTo.Duration != 0 {
+			return fmt.Errorf("no lock gauge must have duration set to 0, was (%d)", m.DistributeTo.Duration)
```

Comment:
nit: I like it when earlier to referred to it as 'no lock', because i keep reading this incorrectly otherwise and it confuses me for a sec

Reply:
changed